app/collectors/reservation.scala

```diff
@@ -50,35 +52,35 @@ case class Reservation(
   duration: Long,
   instanceTenancy: String,
   offeringType: String,
-  startTime: Option[DateTime],
-  endTime: Option[DateTime]
+  startTime: Option[Instant],
+  endTime: Option[Instant]
 ) extends IndexedItem {
   override def callFromArn: (String) => Call = arn => routes.Api.reservation(arn)
 
 }
 
 object Reservation {
   def fromApiData(reservationInstance: ReservedInstances, origin: AmazonOrigin): Reservation = {
-    val region = reservationInstance.getAvailabilityZone
-    val arn = s"arn:aws:ec2:$region:${origin.accountNumber.getOrElse("")}:reservation/${reservationInstance.getReservedInstancesId}"
-    val recurringCharges = reservationInstance.getRecurringCharges.asScala.map(RecurringCharge.fromApiData).toList
+    val region = reservationInstance.availabilityZone
+    val arn = s"arn:aws:ec2:$region:${origin.accountNumber.getOrElse("")}:reservation/${reservationInstance.reservedInstancesId}"
+    val recurringCharges = reservationInstance.recurringCharges.asScala.map(RecurringCharge.fromApiData).toList
     Reservation(
       arn = arn,
-      id = reservationInstance.getReservedInstancesId,
+      id = reservationInstance.reservedInstancesId,
       region = region,
-      instanceType = reservationInstance.getInstanceType,
-      instanceCount = reservationInstance.getInstanceCount,
-      productDescription = reservationInstance.getProductDescription,
-      fixedPrice = reservationInstance.getFixedPrice,
-      usagePrice = reservationInstance.getUsagePrice,
+      instanceType = reservationInstance.instanceTypeAsString,
+      instanceCount = reservationInstance.instanceCount,
+      productDescription = reservationInstance.productDescriptionAsString,
+      fixedPrice = reservationInstance.fixedPrice,
+      usagePrice = reservationInstance.usagePrice,
       recurringCharges = recurringCharges,
-      state = reservationInstance.getState,
-      currencyCode = reservationInstance.getCurrencyCode,
-      duration = reservationInstance.getDuration,
-      instanceTenancy = reservationInstance.getInstanceTenancy,
-      offeringType = reservationInstance.getOfferingType,
-      startTime = Try(new DateTime(reservationInstance.getStart)).toOption,
-      endTime = Try(new DateTime(reservationInstance.getEnd)).toOption
+      state = reservationInstance.stateAsString,
+      currencyCode = reservationInstance.currencyCodeAsString,
+      duration = reservationInstance.duration,
+      instanceTenancy = reservationInstance.instanceTenancyAsString,
+      offeringType = reservationInstance.offeringTypeAsString,
+      startTime = Try(reservationInstance.start).toOption,
+      endTime = Try(reservationInstance.end).toOption
```

Comment:
In the previous implementation the `Try` was needed as `new DateTime` can throw an exception (if the value being passed in isn't a valid DateTime. With the new SDK there is no risk of an exception so we can remove the `Try`. If `start` and `end` can be null then we might need to wrap it with `Option(<value>)` but otherwise we could just pass it through directly and remove the `Option` type from `Reservation`.

Reply:
Look at the number of instances of a field that isn't an option (always there) and our time fields which are optional (will be absent if the date format is bad or the value is null):
```
❯ curl -s https://prism.gutools.co.uk/reservations | jq | grep offeringType | wc -l
     928
❯ curl -s https://prism.gutools.co.uk/reservations | jq | grep startTime | wc -l
     928
❯ curl -s https://prism.gutools.co.uk/reservations | jq | grep endTime | wc -l
     928
```

From this I reckon `startTime` and `endTime` can't be null so it's safe to remove the option entirely.

Reply:
This explanation is amazing! Thank you! 💯👏🏽🥇👍🏽😀